### ISSUE TITLE
request npm package timeout, then process exited

### ISSUE THREAD

**Issue by FeelyChau** [alibaba/pipcook#390]
If request npm package timeout, the daemon process will exit, we should handle this error.
```shell
2020-07-31 17:05:41,185 ERROR 25209 nodejs.RequestError: Error: ESOCKETTIMEDOUT (uncaughtException throw 1 times on pid:25209)
    at new RequestError (/path/pipcook//node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/path/pipcook//node_modules/request/request.js:185:22)
    at Request.emit (events.js:315:20)
    at ClientRequest.<anonymous> (/path/pipcook//node_modules/request/request.js:819:16)
    at Object.onceWrapper (events.js:421:28)
    at ClientRequest.emit (events.js:315:20)
    at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
    at Object.onceWrapper (events.js:421:28)
    at TLSSocket.emit (events.js:327:22)
    at TLSSocket.Socket._onTimeout (net.js:481:8)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
name: "RequestError"
cause: {"code":"ESOCKETTIMEDOUT","connect":false}
error: {"code":"ESOCKETTIMEDOUT","connect":false}
options: {"timeout":15000,"simple":false,"resolveWithFullResponse":true,"uri":"https://registry.npmjs.com/@pipcook/plugins-csv-data-collect","method":"GET","transform2xxOnly":false}
response: undefined
pid: 25209
hostname: MacBook-Pro.local

[Fri Jul 31 2020 17:05:41 GMT+0800 (GMT+08:00)] [graceful:worker:25209:uncaughtException] throw error 1 times
RequestError: Error: ESOCKETTIMEDOUT
    at new RequestError (/path/pipcook//node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/path/pipcook//node_modules/request/request.js:185:22)
    at Request.emit (events.js:315:20)
    at ClientRequest.<anonymous> (/path/pipcook//node_modules/request/request.js:819:16)
    at Object.onceWrapper (events.js:421:28)
    at ClientRequest.emit (events.js:315:20)
    at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
    at Object.onceWrapper (events.js:421:28)
    at TLSSocket.emit (events.js:327:22)
    at TLSSocket.Socket._onTimeout (net.js:481:8)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7) {
  cause: Error: ESOCKETTIMEDOUT
      at ClientRequest.<anonymous> (/path/pipcook//node_modules/request/request.js:816:19)
      at Object.onceWrapper (events.js:421:28)
      at ClientRequest.emit (events.js:315:20)
      at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
      at Object.onceWrapper (events.js:421:28)
      at TLSSocket.emit (events.js:327:22)
      at TLSSocket.Socket._onTimeout (net.js:481:8)
      at listOnTimeout (internal/timers.js:549:17)
      at processTimers (internal/timers.js:492:7) {
    code: 'ESOCKETTIMEDOUT',
    connect: false
  },
  error: Error: ESOCKETTIMEDOUT
      at ClientRequest.<anonymous> (/path/pipcook//node_modules/request/request.js:816:19)
      at Object.onceWrapper (events.js:421:28)
      at ClientRequest.emit (events.js:315:20)
      at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
      at Object.onceWrapper (events.js:421:28)
      at TLSSocket.emit (events.js:327:22)
      at TLSSocket.Socket._onTimeout (net.js:481:8)
      at listOnTimeout (internal/timers.js:549:17)
      at processTimers (internal/timers.js:492:7) {
    code: 'ESOCKETTIMEDOUT',
    connect: false
  },
  options: {
    timeout: 15000,
    simple: false,
    resolveWithFullResponse: true,
    uri: 'https://registry.npmjs.com/@pipcook/plugins-csv-data-collect',
    method: 'GET',
    callback: [Function: RP$callback],
    transform: undefined,
    transform2xxOnly: false
  },
  response: undefined
}
RequestError: Error: ESOCKETTIMEDOUT
    at new RequestError (/path/pipcook//node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/path/pipcook//node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/path/pipcook//node_modules/request/request.js:185:22)
    at Request.emit (events.js:315:20)
    at ClientRequest.<anonymous> (/path/pipcook//node_modules/request/request.js:819:16)
    at Object.onceWrapper (events.js:421:28)
    at ClientRequest.emit (events.js:315:20)
    at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
    at Object.onceWrapper (events.js:421:28)
    at TLSSocket.emit (events.js:327:22)
    at TLSSocket.Socket._onTimeout (net.js:481:8)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
[Fri Jul 31 2020 17:05:41 GMT+0800 (GMT+08:00)] [graceful:worker:25209] will exit after 30000ms
```

**Comment by FeelyChau** [comment]
The reason why the main process exited is: client process throw out an error, controller put the error to the log stream, but no client subscribes it.
Fixed in #380  , when the log manager gets an error, make sure the error event has been subscribed, then emit.

**Comment by FeelyChau** [comment]
It's done by #380.
